app/src/org/commcare/UpdateToPrompt.java

```diff
@@ -0,0 +1,158 @@
+package org.commcare;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.util.Base64;
+
+import org.javarosa.core.model.utils.DateUtils;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.javarosa.core.util.externalizable.ExtUtil;
+import org.javarosa.core.util.externalizable.Externalizable;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.Date;
+
+/**
+ * Created by amstone326 on 4/13/17.
+ */
+
+public class UpdateToPrompt implements Externalizable {
```

Comment:
Did you consider using two separate classes for apkVersion and ccVersion, possibly with a shared super? Not certain that it'd be cleaner but it does seem like there's a good amount of `if`/`else` checking and not that many shared paths.